### DISCUSSION TITLE
Prevent modal from closing outside of the modal

### DISCUSSION THREAD

**Discussion by lucacastelnuovo** [filamentphp/filament#6752] [Help] [answered]
Just allow closing the modal with the cancel button.

**Comment by danharrin** [comment]
Isn't there a config option for this?

**Reply by lucacastelnuovo** [reply]
<img width="1016" alt="image" src="https://github.com/filamentphp/filament/assets/26206253/c8d84ab3-ea1d-4fa4-9020-d90c1c09ed65">
<img width="851" alt="image" src="https://github.com/filamentphp/filament/assets/26206253/3b0ce7f8-80a3-45cd-8c5d-5393df8b7c6b">


In both `config/filament.php` & `config/forms.php` I did not find the option.

**Reply by lucacastelnuovo** [reply]
It seems to be a config option of filament-support.

[vendor/filament/support/src/Actions/Concerns/CanOpenModal.php](https://github.com/filamentphp/filament/blob/2.x/packages/support/config/filament-support.php)
```php
// ...

public function isModalClosedByClickingAway(): bool
    {
        return $this->evaluate($this->isModalClosedByClickingAway) ?? config('filament-support.modal.is_closed_by_clicking_away') ?? true;
    }

// ...
```

**Comment by pboivin** [comment, marked as answer]
@Luca-Castelnuovo Try this on your action:
```
    ->closeModalByClickingAway(false)
```

**Reply by lucacastelnuovo** [reply]
Awesome, this fixes the problem.

**Reply by valpuia** [reply]
> @Luca-Castelnuovo Try this on your action:
> 
> ```
>     ->closeModalByClickingAway(false)
> ```

What about "Esc" key.. Is there a way to prevent also

**Reply by pboivin** [reply]
I don't think there's an option for this one unfortunately

**Reply by carlosrl19** [reply]
> > @Luca-Castelnuovo Try this on your action:
> > ```
> >     ->closeModalByClickingAway(false)
> > ```
> 
> What about "Esc" key.. Is there a way to prevent also

I think is too late to an answer but if you need to prevent modal close by ESC key you can use ```->closeModalByEscaping(false)``` (Filament v3.x)

**Comment by abbasmashaddy72** [comment]
Modal::closedByClickingAway(false) is not working tried on middleware and service provider


https://github.com/filamentphp/filament/assets/55267488/a326ead0-34b8-4eb9-8d46-035348252853

**Reply by lucacastelnuovo** [reply]
This issue was originally created for Filament V2. Which version are you running?

**Reply by abbasmashaddy72** [reply]
I got it solved No issues
